### ISSUE TITLE
students are generally unclear about the pair-programming logistics

### ISSUE THREAD

**Issue by bill-auger** [saasbook/AV102#43]
to the TAs -

i think we really need to attend to the student gitter chat room better at least for the first few weeks to help students co-ordinate pair-programming and it seems that so far i have been the only TA regularly in the chat - have seen sam a few times also but i think we really should strive for 24 hr coverage here

the typical things i am seeing nearly every hour is those who are not sure of the basics of pair programming - like: "what is expected in pair programming?", "will someone pair with me right now?", "is anyone in mexico?" - the last type implying (.. to be my one and only permanent partner) - this is an especially discouraging mis-conception because finding one and only one permanent partner for the entire course who will be available for you at all times you are available is obviously unreasonable but this seems to be the general perception - and if students are not using the agile ventures events page then this is going to be nearly impossible for them

i am repeatedly telling them to read the "pair programming" page and to search for or create events on agileventures well in advance - that these events will be listed for others to see and links posted in the chat when they begin

and i find myself repeatedly emphasizing that one should not expect to find one person who will be available for them always - that it is in their interest to find someone different each time - perhaps several per week

i have looked over the site quite a bit and this all seems quite clear to me but many students seem to be missing these key aspects and because pair-programming is a strict requirement of the course now we really need to work harder to ensure student find each other - perhaps sam could add yet another video making these points more clear but again IMHO they are already - i just dont know how else we can make these basic things more clear other than catching them in the chat room (and i mean actively "catching" them becuase the usually leave the chat after 5 minutes of no response)

**Comment by bill-auger** [comment]
for reference i have these snippets of typical advice saved into my clip manager:
- the location of your pair partner is not at all important - as long as two students are online at the same time that is all that is required
- you are not required to pair with the same person each time - in fact you are encouraged to find someone new each time
- just go to the agileventures events page and create an event scheduled for a few days in the future at your preferred time - that will remain on the list until it begins - also look at the list yourself to see if someone else has already posted an event for some time that is good for you and simply plan to join them when it begins - a link into the hangout will be posted on the event page and also in the gitter chat when the event begins
- the "Pair Programming" instructions are here --> https://courses.edx.org/courses/course-v1:BerkeleyX+CS169.1x+3T2015SP/77935ad32a3943d6b7a177065233358d/
- also you could simply stay in this chat and all events are posted with links here in this chat - you are welcome to join any that you see
- everyone is welcome to join and observe any session - even if you only watch - also there is generally nothing wrong with having multiple navigators - as long as there is a single driver this usually works out just as well - but you should try to be the driver yourself for the session that you submit as your assignment

**Comment by armandofox** [comment]
i agree we need a way to coordinate more spontaneous pairing. 

maybe announce "pair dating" times on a permanently-open hangout, so that (eg) if people show up on any even-numbered hour, they can find pair(s) near them?  could do hangouts per country/region too if that works better.

i agree we want to encourage promiscuity in pairing too.  did these students watch the pair programming lectures?  we were pretty clear that there is a specific procedure for pairing, and that pairs should switch roles often and people should switch pairs often....

**Comment by tansaku** [comment]
@bill-auger thanks for flagging this.  I think we do have it covered in the videos - possibly we need to set up a faq page at the end of the main pairing page so that we can specifically refer people to that ...

I'm all ears for ideas on better coordination - we could be surveying students on timezone availability etc. and doing more to assign partners ... maybe assign people into groups of four with the assumption that half will drop - or say that assigned partners is something you get as part of the verified certificate package ...?

**Comment by tansaku** [comment]
@bill-auger - I added a FAQ to the bottom of https://courses.edx.org/courses/course-v1:BerkeleyX+CS169.1x+3T2015SP/77935ad32a3943d6b7a177065233358d/ - based on your notes - please see my slight changes - feel free to pin links of these - copies of these etc. in appropriate places

**Comment by bill-auger** [comment]
after observing a few pair sessions today i suspect that many have simply not seen the "pair programming" page or even looked at the assignment before looking for a partner or blindly jumping into an existing pairing session - only then if they get that far asking or figuring out what they are supposed to do - for example in one session i observed the first guy was well prepared and created the hangout with no help and his partner was ill prepared and had not even looked at the assignment from this week nor last week's

so i am even more convinced now that the courseware is already sufficiently clear but some students are going to rush through missing key points and the only way to reach them is directly in the chat and hangouts at the precise moment they stumble on something before they give up and/or go it solo

**Comment by tansaku** [comment]
yeah, some people will always just leap in unprepared - not sure what we can do about that ... apart from some more structured system where people have done a pairing precourse before they are allowed to participate ...
